sql/core/src/main/scala/org/apache/spark/sql/execution/DataSourceScanExec.scala

```diff
@@ -313,8 +321,10 @@ case class FileSourceScanExec(
   override lazy val metrics =
     Map("numOutputRows" -> SQLMetrics.createMetric(sparkContext, "number of output rows"),
       "numFiles" -> SQLMetrics.createMetric(sparkContext, "number of files"),
+      "scanTime" -> SQLMetrics.createTimingMetric(sparkContext, "scan time"),
       "fileListingTime" -> SQLMetrics.createMetric(sparkContext, "file listing time (ms)"),
```

Comment:
Currently, the file listing time still contains part of partition pruning time(introduced from [SPARK-20151](https://github.com/apache/spark/pull/17476/files)), I think we should split the partition pruning time into `metastore operation time`, which will be done in https://issues.apache.org/jira/browse/SPARK-26223.